ang/civicase-base.ang.php

```diff
@@ -0,0 +1,185 @@
+<?php
+
+/**
+ * @file
+ * Declares an Angular module which can be autoloaded in CiviCRM.
+ *
+ * See also:
+ * http://wiki.civicrm.org/confluence/display/CRMDOC/hook_civicrm_angularModules.
+ */
+
+use Civi\CCase\Utils as Utils;
+
+$options = [
+  'activityTypes' => 'activity_type',
+  'activityStatuses' => 'activity_status',
+  'caseStatuses' => 'case_status',
+  'priority' => 'priority',
+  'activityCategories' => 'activity_category',
+  'caseTypeCategories' => 'case_type_categories',
+];
+
+set_option_values_to_js_vars($options);
+set_case_types_to_js_vars($options);
+set_relationship_types_to_js_vars($options);
+set_file_categories_to_js_vars($options);
+set_activity_status_types_to_js_vars($options);
+set_custom_fields_info_to_js_vars($options);
+set_tags_to_js_vars($options);
+
+if (!function_exists('glob_recursive')) {
+
+  /**
+   * Recursive Glob function.
+   *
+   * Source: http://php.net/manual/en/function.glob.php#106595
+   * Does not support flag GLOB_BRACE.
+   */
+  function glob_recursive($pattern, $flags = 0) {
+    $files = glob($pattern, $flags);
+
+    foreach (glob(dirname($pattern) . '/*', GLOB_ONLYDIR | GLOB_NOSORT) as $dir) {
+      $files = array_merge($files, glob_recursive($dir . '/' . basename($pattern), $flags));
+    }
+
+    return $files;
+  }
+
+}
```

Comment:
This function will be used by all dependent extentions(awards/prospects etc). Will it work if I move it to a separate file?
As this is a base module, if I write a function here, it will be globally available. So I feel this is the way to go. But again, I am no expert in PHP.